Dockerfile

```diff
@@ -0,0 +1,66 @@
+################################################################
+################################################################
+################################################################
+
+ARG DEBIAN_VERSION=bookworm
+
+################################################################
+
+ARG NODE_VERSION=hydrogen
+ARG NODE_CONTAINER_TAG=${NODE_VERSION}-${DEBIAN_VERSION}-slim
+
+################################################################
+################################################################
+################################################################
+
+FROM node:${NODE_CONTAINER_TAG} as builder
+
+################################################################
+
+USER root
+RUN npm i -g npm
+RUN npm i -g yarn --force
+
+################################################################
+
+WORKDIR /srv
+
+################################################################
+
+RUN chown -R node:node /srv/ && \
+    chmod 0750 /srv/
+
+COPY --chown=node:node package.json yarn.lock /srv/
+RUN chmod 0640 /srv/package.json /srv/yarn.lock
+
+################################################################
+
+USER node
+RUN yarn install --frozen-lockfile
+
+################################################################
+
+COPY --chown=node:node . /srv
+RUN yarn build
```

Comment:
Appropriate env variables are required for a production build. `yarn build` by default looks for env variables from `.env.production`. As these are not configured the site doesn't know what wss url to connect to and so errors when you try to connect a wallet. To use an alternate env variable file you can do, for example, `yarn build --mode testnet` or `yarn build --mode mainnet` which would pull variables from either `.env.testnet` or `.env.mainnet`

Reply:
updated with https://github.com/PolymeshAssociation/polymesh-portal/pull/23/commits/5adb540f28c3c9bbee2283a3f1650ac112fb0673
```
adds dynamic env management; template values are set in the form `ENV_VAR_NAME` = `__ENV_VAR_NAME__`, which are replaced at runtime with the container's set environment values - this allows the container to be portable/reusable, and enables consistent code/container state throughout the various environments
```